src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.Unix.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+using System.Buffers;
+using System.IO;
+using System.Reflection;
+
+namespace System.Runtime.Loader
+{
+    public partial class AssemblyLoadContext
+    {
+        // Find satellite path using case insensitive culture name
+        internal static unsafe string? FindCaseInsensitiveSatellitePath(string parentDirectory, string cultureName, string assembly)
+        {
+            int bufferSize = Interop.Sys.GetReadDirRBufferSize();
+            byte[]? dirBuffer = null;
+            try
+            {
```

Comment:
why don't move this try after ArrayPool<byte>.Shared.Rent(bufferSize); and then avoid checking for null in the finally block?

Reply:
I assumed the only purpose of the outer try block was to catch the case when `ArrayPool<byte>.Shared.Rent(bufferSize)` threw.